src/components/Form/FormFormats.js

```diff
@@ -0,0 +1,792 @@
+                    [FORM_ERROR]: response.data,
+                };
+            }
+        },
+    },
+    "parent": {
+        "title": "Parent",
+        "form": [PARENT_FIELDS],
+        // TODO: loading and submitting with GraphQL
+        "load": (id) => {},
+        "submit": async (dispatch, formData, id) => {},
+    },
+    "admin": {
+        "title": "Administrator",
+        "form": [
+            {
+                "name": "login",
+                "label": "Login Details",
+                "fields": [
+                    EMAIL_FIELD,
+                    {
+                        "name": "password",
+                        "label": "Password",
+                        "component": <Fields.TextField type="password" />,
+                        "validator": Yup.mixed(),
+                        "required": true,
+                    },
+                    ...NAME_FIELDS,
+                ],
+            },
+            {
+                "name": "user",
+                "label": "User Information",
+                "fields": [
+                    {
+                        "name": "adminType",
+                        "label": "Admin Type",
+                        ...selectField([
+                            {"label": "Owner", "value": "OWNER"},
+                            {"label": "Receptionist", "value": "RECEPTIONIST"},
+                            {"label": "Assistant", "value": "ASSISTANT"},
+                        ]),
+                        "required": true,
+                    },
+                    GENDER_FIELD,
+                    BIRTH_DATE_FIELD,
+                    ADDRESS_FIELD,
+                    CITY_FIELD,
+                    PHONE_NUMBER_FIELD,
+                    STATE_FIELD,
+                    ZIPCODE_FIELD,
+                ],
+            },
+        ],
+        // TODO: add administrator loading
+        "load": async (id) => {
+            const GET_ADMIN = gql`
+            query GetAdmin($userID: ID!) {
+                admin(userId: $userID) {
+                    user {
+                    id
+                    email
+                    firstName
+                    lastName
+                    }
+                    adminType
+                    gender
+                    phoneNumber
+                    birthDate
+                    address
+                    city
+                    state
+                    zipcode
+                }
+            }`;
+            try {
+                const {"data": {admin}} = await client.query({
+                    "query": GET_ADMIN,
+                    "variables": {
+                        "userID": id,
+                    },
+                });
+                return {
+                    "login": admin.user,
+                    "user": admin,
+                };
+            } catch (error) {
+                return null;
+            }
+        },
+        "submit": async (formData, id) => {
```

Comment:
I see what you're doing here... you're defining all of the submit and load actions here... I think as a short-term solution this is fine, but in the not so distant future, we want to dynamically generate these queries based on the graphQL schema. This is because we eventually want to support customizable forms (an admin can add/delete additional fields). 

For this, we'll likely need the backend to generate the graphQL query and return it to us when we make our graphql request to that schema. Here's a [library](https://github.com/timqian/gql-generator) that someone made that does this but it's for API testing. This is also an [interesting walk through](https://itnext.io/building-a-dynamic-controlled-form-in-react-together-794a44ee552c) of a similar type of form we're trying to build.I think this would be a good project for you to work on for Mainframe - storing custom form fields and generating the necessary query/mutation strings.